jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdCredentials.java

```diff
@@ -224,40 +223,23 @@ and the Token Endpoint (which it is in this flow), the TLS server validation
         return paddedEncodedJwtSection;
     }
 
-    private Map<String, Object> claimAuthCode(String authCode) throws IOException
+    private Map<String, Object> claimAuthCode(HttpClient httpClient, String authCode) throws Throwable
     {
+        Fields fields = new Fields();
+        fields.add("code", authCode);
+        fields.add("client_id", configuration.getClientId());
+        fields.add("client_secret", configuration.getClientSecret());
+        fields.add("redirect_uri", redirectUri);
+        fields.add("grant_type", "authorization_code");
+        FormContentProvider formContentProvider = new FormContentProvider(fields);
+        Request request = httpClient.POST(configuration.getTokenEndpoint())
```

Comment:
I would add the try/catch back and add the clear logging messages about what went wrong. Was it better to throw for some reason?

```
private Map<String, Object> claimAuthCode(HttpClient httpClient, String authCode)
    {
        Map<String, Object> result;

        try
        {
            ...
        }
        catch (InterruptedException e)
        {
            LOG.debug("Call to token endpoint was interrupted", e);
            result = Collections.emptyMap();
        }
        catch (ExecutionException e)
        {
            LOG.warn("Call to token endpoint was failed", e);
            result = Collections.emptyMap();
        }
        catch (TimeoutException e)
        {
            LOG.warn("Call to token endpoint was did not complete in a timely manner", e);
            result = Collections.emptyMap();
        }

        return result;
    }
```

Reply:
With that code we are still throwing, we have just delayed the throw until we go out to `redeemAuthCode`, and then throw because we have no `id_token` which is not the reason of the failure.

Reply:
That's a point.